src/jaxsim/api/data.py

```diff
@@ -43,6 +43,24 @@ class JaxSimModelData(common.ModelDataWithVelocityRepresentation):
         default_factory=lambda: jnp.array(0, dtype=jnp.uint64)
     )
 
+    def __hash__(self) -> int:
+
+        return hash(
+            (
+                hash(self.state),
+                hash(tuple(self.gravity.flatten().tolist())),
+                hash(self.soft_contacts_params),
+                hash(jnp.atleast_1d(self.time_ns).flatten().tolist()),
```

Comment:
I guess that `time_ns` has shape `()`, so probably a `.tolist()` of a `.item()` should be enough
```suggestion
                hash(self.time_ns.tolist()),
```

Reply:
The operator `jnp.atleast1d(X).flatten()` is there mainly for making the hash function work on vectorized pytrees.